src/topology/metric_space/basic.lean

```diff
@@ -1876,6 +1876,20 @@ metric_space.induced coe (λ x y, subtype.ext) t
 
 theorem subtype.dist_eq {p : α → Prop} (x y : subtype p) : dist x y = dist (x : α) y := rfl
 
+instance : metric_space empty :=
+{ dist := λ _ _, 0,
+  dist_self := λ _, rfl,
+  dist_comm := λ _ _, rfl,
+  eq_of_dist_eq_zero := λ _ _ _, subsingleton.elim _ _,
+  dist_triangle := λ _ _ _, show (0:ℝ) ≤ 0 + 0, by rw add_zero, }
+
+instance : metric_space punit :=
```

Comment:
Should we have the same for `pempty` and `unit`? It's really a shame that there isn't more defeq here...

Reply:
`unit` is an abbreviation for `punit`, so this instance should also work for `unit` (this is unfortunately not the case for `empty`/`pempty`)

Reply:
An option for `empty`/`pempty` is to merge #7606 first and add an instance `is_empty -> metric_space`.

Reply:
I vote just punting on this.